backend/messaging-api/messaging-api.yaml

```diff
@@ -0,0 +1,190 @@
+          type: string
+        environmentId:
+          x-basename: eId
+          type: string
+          format: uuid
+        portfolioId:
+          x-basename: pId
+          type: string
+          format: uuid
+        organizationId:
+          x-basename: oId
+          type: string
+          format: uuid
+        applicationId:
+          x-basename: aId
+          type: string
+          format: uuid
+        whenUpdated:
+          type: string
+          format: date-time
+        whoUpdated:
+          type: string
+          description: Name of the user who made the change to the feature
+        featureKey:
+          description: "Feature key that got updated"
+          type: string
+        featureValueUpdated:
+          description: "Feature value changes"
+          $ref: '#/components/schemas/MessagingFeatureValueUpdate'
+        lockUpdated:
+          description: "Lock changes"
+          $ref: '#/components/schemas/MessagingLockUpdate'
+        retiredUpdated:
+          description: "Retired changes"
+          $ref: '#/components/schemas/MessagingRetiredUpdate'
+        strategiesUpdated:
+          description: "Strategies changes"
+          type: array
+          items:
+            $ref: '#/components/schemas/MessagingStrategyUpdate'
+        strategiesReordered:
+          description: "Strategies reorder changes"
+          $ref: '#/components/schemas/MessagingStrategiesReorder'
+        additionalInfo:
+          description: "Any additional info, webhook headers etc"
+          type: array
+          items:
+            type: string
+
+    MessagingFeatureValueUpdate:
+      x-package: io.featurehub.messaging.model
+      type: object
+      description: "Feature value changes, if any"
+      required:
+        - valueType
+        - updated
+        - previous
+      properties:
+        valueType:
+          $ref: "#/components/schemas/FeatureValueType"
+        updated:
+          description: "Updated feature value"
+        previous:
+          description: "Previous feature value"
+    MessagingLockUpdate:
+      x-package: io.featurehub.messaging.model
+      type: object
+      description: "Lock value changes, if any"
+      required:
+        - updated
+        - previous
+      properties:
+        updated:
+          type: boolean
+          description: "Updated lock value"
+        previous:
+          type: boolean
+          description: "Previous lock value"
+    MessagingRetiredUpdate:
+      x-package: io.featurehub.messaging.model
+      type: object
+      description: "Retired value changes, if any"
+      required:
+        - updated
+        - previous
+      properties:
+        updated:
+          type: boolean
+          description: "Updated retired value"
+        previous:
+          type: boolean
+          description: "Previous retired value"
+
+    MessagingStrategiesReorder:
```

Comment:
just in case i forget, i was wondering if an add of a new strategy to the end causes a reorder? It would seem unnecessary to create this data for that reason. I just put it here to remember.